### ISSUE TITLE
VCF input error: "TypeError: startswith first arg must be bytes or a tuple of bytes, not str"

### ISSUE THREAD

**Issue by beausoleilmo** [isaacovercast/easySFS#26]
I'm trying to run the command, but there is a problem with startswith (I assume that this is the header of the VCF, which in my case is present and has a bunch of "#" at the beginning)

```
easySFS.py -i output.vcf.gz -p $VOLUME/sfs/pop2.list -a -f --preview
Processing 1 populations - odict_keys(['Species_site-Site'])
Error - startswith first arg must be bytes or a tuple of bytes, not str
Traceback (most recent call last):
  File "easySFS.py", line 646, in <module>
    main()
  File "easySFS.py", line 590, in main
    indnames = get_inds_from_input(args.vcf_name, args.verbose)
  File "easySFS.py", line 391, in get_inds_from_input
    if line.startswith('#'):
TypeError: startswith first arg must be bytes or a tuple of bytes, not str
```

When I look at the #CHROM in the VCF, I see this: 
```
bcftools view -h output.vcf.gz | grep -i -B 1 -A 1 '#CHROM'
```
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  ID1      ID2      ID3      ID4      ID5

I compared with the file that was provided in the example and it seems similar... 

What could cause that and how to fix it? 

The VCF is v4.2
```
##fileformat=VCFv4.2
##FILTER=<ID=PASS,Description="All filters passed">
##samtoolsVersion=1.10+htslib-1.10.2
```

**Comment by isaacovercast** [comment]
What version of python are you running? I will guess 2.7. I just converted some of the code to 2.7/3 compatibility and some of it might not be perfect yet. Can you post the full vcf header plus the first line or 2 of the data?

**Comment by beausoleilmo** [comment]
Here is what I can find: 

```
python --version
Python 2.7.17
```

```
##fileformat=VCFv4.2
##FILTER=<ID=PASS,Description="All filters passed">
##samtoolsVersion=1.10+htslib-1.10.2
##samtoolsCommand=samtools mpileup -C 50 -E -t SP -t DP -u -I -f pathto/ref_gen/UCSC/refg.fa -b pathto/bam_lists/bam_list.txt
##reference=file://pathto/ref_gen/UCSC/refg.fa
##contig=<ID=JH739887,length=30495534>
```

```
##bcftools_viewCommand=view -h output.vcf.gz; Date=Tue Feb 18 16:44:38 2020
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  sample1      sample2      sample3      sample4      sample5
scaff1        266     .       A       C       142.253 .       DP=27;VDB=0.195169;SGB=5.93809;RPB=0.0923061;MQB=0.317882;MQSB=0.95891;BQB=0.19945;MQ0F=0;AF1=0.416951;AC1=4;DP4=6,4,2,9;MQ=33;FQ=144.609;PV4=0.0804954,0.0240713,0.107328,1    GT:PL:DP:SP:GQ  0/1:0,0,0:0:0:3 0/0:0,6,76:2:0:6        1/1:95,15,0:5:0:11      0/0:0,9,74:3:0:8        0/1:94,0,114:11:12:95
scaff1        267     .       C       T       109.807 .       DP=27;VDB=0.219282;SGB=5.17124;RPB=0.0642024;MQB=0.250861;MQSB=0.999215;BQB=0.416862;MQ0F=0;AF1=0.405763;AC1=4;DP4=6,4,1,9;MQ=33;FQ=112.065;PV4=0.0572755,0.0396409,0.07164,1   GT:PL:DP:SP:GQ  0/1:0,0,0:0:0:3 0/0:0,6,76:2:0:6        1/1:62,12,0:4:0:8       0/0:0,9,74:3:0:9        0/1:94,0,114:11:12:95
```

**Comment by isaacovercast** [comment]
This is fixed: bdcd9c0. Go ahead and pull down a new copy of the code from the repo and try again.

The problem was the file was being opened in binary mode since it was gzipped. It always should have allowed gzipped vcf, but now it _really_ does ;p

**Comment by beausoleilmo** [comment]
I reinstalled everything like advised in the README. Works like a charm. Thanks a lot for that!
